### ISSUE TITLE
Editable p-select Accessibility Issues

### ISSUE THREAD

**Issue by jongmin-r-kim** [primefaces/primeng#18963]
### Describe the bug

If we have an **editable** p-select element that is initialised to one of the values in the default list, and is labelled by something else, we result in the following accessibility issue:
    `Accessible name does not match or contain the visible label text`
This violations goes away after entering an arbitrary value.

On the other hand, if we lack a label associated with the p-select, the opposite situation occurs; there are no violations if the select list is set to a value from the default list, but we get the following violation if enter a non-default value:
    `Form control element <input> has no associated label`

These violations can be viewed via the  IBM Equal Accessibility Checker. 

### Reason for not contributing a PR

- [x] Lack of time
- [ ] Unsure how to implement the fix/feature
- [ ] Difficulty understanding the codebase
- [ ] Other

### Reproducer

https://stackblitz.com/edit/stackblitz-starters-2qqykztk?file=src%2Fapp%2Fselect.component.html

### Environment

Stackblitz

### Angular version

20.0.4

### PrimeNG version

v20

### Steps to reproduce the behavior

1. Install the [IBM Equal Accessibility Checker ](https://chromewebstore.google.com/detail/ibm-equal-access-accessib/lkcagbfjnkomcinoddgooolagloogehp)
2. Open the stackblitz
3. Change the values for each of the p-select elements while running the accessiblity checker when they have their default/non-default values
4. Filter on violations only and note the following errors:

`Accessible name does not match or contain the visible label text`
`Form control element <input> has no associated label`

### Expected behavior

No accessibility violations on the p-select are present, whether entering arbitrary values or using one of the values from the list.

**Comment by cetincakiroglu** [comment]
fixed for v22.2.0
